File: entrega2/menu/src/main/java/com/aps/menu/entidades/Restaurante.java
package com.aps.menu.entidades;

import javax.persistence.Entity;

@Entity
public class Restaurante extends Conta {
    private String nome;
    private String endereco; // MUDAR PARA CLASSE ENDERECO
    private String telefone;


    public Restaurante(Long id, String email, String senha, String nome, String endereco, String telefone) {
        super(id, email, senha);
        this.nome = nome;
        this.endereco = endereco;
        this.telefone = telefone;
    }

    public Restaurante(Long id, String email, String senha) {
        super(id, email, senha);
    }

    public Restaurante () {
        
    }

}

File: soa/menu-app/acesso-service/src/entidades/restaurante.ts
import { Conta } from "./conta";
import { Endereco } from "./endereco";

export class Restaurante extends Conta {
    private nome: string;
    private endereco: Endereco;
    private telefone: string;

    constructor(id: number, email: string, senha: string, nome: string, endereco: Endereco, telefone: string) {
        super(id, email, senha);
        this.nome = nome;
        this.endereco = endereco;
        this.telefone = telefone;
    }

    public getNome() : string {
        return this.nome;
    }
    
    public getEndereco() : Endereco {
        return this.endereco;
    }

    public getTelefone() : string {
        return this.telefone;
    }
}